src/lib.rs

```diff
@@ -302,38 +302,85 @@ impl Database {
     /// It will simply skip malformed fonts and will print a warning into the log for each of them.
     #[cfg(feature = "fs")]
     pub fn load_fonts_dir<P: AsRef<std::path::Path>>(&mut self, dir: P) {
-        self.load_fonts_dir_impl(dir.as_ref())
+        self.load_fonts_dir_impl(dir.as_ref(), &mut Default::default())
```

Comment:
Shouldn't we init `HashSet` in `load_system_fonts`? Right now we would have to populate it each time `load_fonts_dir` hits a symlink.

And we can add private `load_fonts_dir_impl` that accepts a `HashSet`. So the public API would not change.

Reply:
`load_system_fonts` uses `load_fonts_dir`, not `load_fonts_dir_impl`. So HashMap is being initialized multiple times.

Reply:
I see, I changed the other callsites.